### PR TITLE
Expose articulation motor parameters, fixed motor creation in articul…

### PR DESCRIPTION
**Pull request by michalpelka**
…ation creation code.

## What does this PR do?

Adds articulation's motor parameter, correctly setup motor.

## How was this PR tested?

Manual tested.

### CODE REVIEW

File: Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp [hunk 1]
```diff
@@ -376,6 +385,15 @@ namespace PhysX
...
                         physx::PxArticulationAxis::eX, physx::PxArticulationMotion::eFREE); // free on the x movement axis (eX)
                 }
+                if (articulationLinkConfiguration.m_motorConfiguration.m_useMotor)
+                {
+                    physx::PxArticulationDrive drive;
+                    drive.driveType = physx::PxArticulationDriveType::eFORCE;
+                    drive.maxForce = articulationLinkConfiguration.m_motorConfiguration.m_driveForceLimit;
+                    drive.damping = articulationLinkConfiguration.m_motorConfiguration.m_driveDamping;
+                    drive.stiffness = articulationLinkConfiguration.m_motorConfiguration.m_driveStiffness;
+                    inboundJoint->setDriveParams(physx::PxArticulationAxis::eTWIST, drive);
```
Comment (moraaar) [inline Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp:395]: Is the right axis `eTWIST` (which is rotation) for prismatic? Normally prismatic only allows movement in the x axis, so how the limits apply to `eX` axis.

Comment (michalpelka) [reply Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp]: Fixed, thanks!

File: Gems/PhysX/Code/Source/Articulation/ArticulationLinkConfiguration.h [hunk 2]
```diff
@@ -41,6 +41,21 @@ namespace PhysX
...
+    //! Configuration articulation motor
+    struct ArticulationJointMotorProperties
+    {
+        AZ_CLASS_ALLOCATOR_DECL;
+        AZ_TYPE_INFO(ArticulationJointMotorProperties, "{44AAC055-73A0-4835-B9E9-A9E233B2C54A}");
+        static void Reflect(AZ::ReflectContext* context);
+
+        ArticulationJointMotorProperties() = default;
+
+        bool m_useMotor = false; ///< Enables joint actuation.
```
Comment (adamdbrw) [inline Gems/PhysX/Code/Source/Articulation/ArticulationLinkConfiguration.h:53]: Inline comments:
`//!<`

File: Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp [hunk 3]
```diff
@@ -376,6 +385,15 @@ namespace PhysX
                     inboundJoint->setMotion(
                         physx::PxArticulationAxis::eX, physx::PxArticulationMotion::eFREE); // free on the x movement axis (eX)
                 }
+                if (articulationLinkConfiguration.m_motorConfiguration.m_useMotor)
+                {
+                    physx::PxArticulationDrive drive;
+                    drive.driveType = physx::PxArticulationDriveType::eFORCE;
+                    drive.maxForce = articulationLinkConfiguration.m_motorConfiguration.m_driveForceLimit;
+                    drive.damping = articulationLinkConfiguration.m_motorConfiguration.m_driveDamping;
+                    drive.stiffness = articulationLinkConfiguration.m_motorConfiguration.m_driveStiffness;
```
Comment (adamdbrw) [inline Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp:394]: This block is repeated, refactor into an internal function:
void setInboundJointDriveParams(inboundJoint, articulationAxis, ArticulationJointMotorProperties).

**Review by adamdbrw — APPROVED** [review]

**Review by greerdv — APPROVED** [review]

File: Gems/PhysX/Code/Source/Pipeline/PhysicsPrefabProcessor.cpp [hunk 4]
```diff
@@ -71,7 +71,7 @@ namespace PhysX
             else
             {
                 linkData->m_jointLeadLocalFrame = AZ::Transform::CreateFromQuaternionAndTranslation(
-                    AZ::Quaternion::CreateFromEulerAnglesDegrees(articulationLinkComponent->m_config.m_leadLocalPosition),
+                    AZ::Quaternion::CreateFromEulerAnglesDegrees(articulationLinkComponent->m_config.m_LeadLocalRotation),
```
Comment (moraaar) [inline Gems/PhysX/Code/Source/Pipeline/PhysicsPrefabProcessor.cpp:74]: Ups, great find!

**Pull request merged**